partner_firstname/tests/test_empty.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2014-2015 Grupo ESOC <www.grupoesoc.es>
+# Copyright 2016 Yannick Vaucher (Camptocamp)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+"""Test situations where names are empty.
+
+To have more accurate results, remove the ``mail`` module before testing.
+"""
+from odoo.tests.common import TransactionCase
+
+from .. import exceptions as ex
+from .base import MailInstalled
+
+
+class CompanyCase(TransactionCase):
+    """Test ``res.partner`` when it is a company."""
+
+    model = "res.partner"
+    context = {"default_is_company": True}
+
+    def tearDown(self):
+        try:
+            data = {"name": self.name}
+            # pylint: disable=W8121
+            model = self.env[self.model].with_context(self.context)
```

Comment:
with_context(**self.context) to avoid the pylint check issue more than disabling the check, or even directly removing the with_context. Why is it needed here?

Reply:
Will try to make it work with kwargs. Again, this came with the original source code.

Reply:
That's not a reason to keep. On every migration we should improve things if possible